### PR TITLE
Cluster with network id - fix

### PR DESCRIPTION
**Pull request by 08volt**
Supporting cluster with network defined using network Id instead of network Name

### CODE REVIEW

File: pkg/network/network.go [hunk 1]
```diff
@@ -61,6 +97,21 @@ func NewNetworksResolver(networkLister, gkeNetworkParamSetLister cache.Indexer,
...
 
+func GetNetworkSelfLink(gceCloud *gce.Cloud) (string, error) {
+	if gceCloud == nil {
+		return "", fmt.Errorf("Netowrk resolver: provided cloud is nil")
+	}
+	url := gceCloud.NetworkURL()
+	lastIndex := strings.LastIndex(url, "/")
+	networkId := url[lastIndex+1:]
+
+	networkResource, err := gceCloud.GetNetwork(networkId)
```
Comment (mmamczur) [inline pkg/network/network.go:108]: can we try to make this fall back to the old behavior if this call fails for whatever reason and log an error?

but maybe we don't need this, up to you

File: pkg/network/network.go [hunk 2]
```diff
@@ -45,13 +46,48 @@ type Resolver interface {
...
 	logger                   klog.Logger
 }
 
+type CloudNetworkProviderAdapterWithSelfLink struct {
+	cloud      *gce.Cloud
+	networkURL string
+}
+
+// CloudNetworkProviderAdapterWithSelfLink uses the network selfLink as networkUrl.
+func NewAdapterNetworkSelfLink(gceCloud *gce.Cloud) (CloudNetworkProvider, error) {
```
Comment (mmamczur) [inline pkg/network/network.go:60]: could you add a test to at least verify if this adapter correctly resolves the network? You can use fakeCloud for this probably

File: pkg/network/network.go (hunk 1 above)
Comment (08volt) [reply pkg/network/network.go:108]: Yes, this is what it does if 
``` 
adapter, err := network.NewAdapterNetworkSelfLink(ctx.Cloud)
```
fails, adapter = ctx.Cloud, which is like it was in the past

Comment (mmamczur) [reply pkg/network/network.go:108]: ah true I haven't noticed this

File: pkg/network/network.go (hunk 2 above)
Comment (08volt) [reply pkg/network/network.go]: fixed

**Comment by swetharepakula** [comment]
What was the context behind needing this change? Could we consider using a utils function instead to convert the URL? I feel this adds unnecessary complexity.

**Comment by mmamczur** [comment]
so there is this issue that when you create a cluster and specify an URL of the VPC to use. If in that scenario you give it an URL with network number instead of the typical network URL with the network name, the cluster will be created and will seemingly work. ingress-gce however, will have this numeric ID in the config. The problem is that if you pass the network URL in this form to some parts of GCE API, it will error on it. So you need to always use the network name, if you only have the number you actually have to do an API call to get the name.

**Pull request merged**